core/app/models/spree/line_item.rb

```diff
@@ -118,14 +116,23 @@ def options=(options = {})
 
       assign_attributes options
 
-      # There's no need to call a pricer if we'll set the price directly.
+      # When price is part of the options we are not going to fetch
+      # it from the variant. Please note that this always allows to set
+      # a price for this line item, even if there is no existing price
+      # for the associated line item in the order currency.
       unless options.key?(:price) || options.key?('price')
-        self.money_price = variant.price_for(pricing_options)
+        self.money_price = price_from_options
       end
     end
 
-    def pricing_options
```

Comment:
I like removing things from the public interface, but I'm not sure about moving this method to the private section.